homeassistant/components/saj/sensor.py

```diff
@@ -0,0 +1,195 @@
+"""SAJ solar inverter interface."""
+import asyncio
+from datetime import date, timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_MONITORED_CONDITIONS,
+    CONF_SCAN_INTERVAL,
+    STATE_UNKNOWN,
+)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_time_interval
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSOR_ICON = "mdi:solar-power"
+
+
+def _check_sensor_schema(conf):
+    """Check sensors and attributes are valid."""
+    try:
+        import pysaj
+
+        valid = [s.name for s in pysaj.Sensors()]
+    except (ImportError, AttributeError):
+        return conf
+
+    for sensor in conf[CONF_MONITORED_CONDITIONS]:
+        if sensor not in valid:
+            raise vol.Invalid(f"{sensor} does not exist")
+    return conf
+
+
+PLATFORM_SCHEMA = vol.All(
+    PLATFORM_SCHEMA.extend(
+        {
+            vol.Required(CONF_HOST): cv.string,
+            vol.Optional(CONF_MONITORED_CONDITIONS, default=[]): vol.Any(
+                vol.All(cv.ensure_list, [str])
+            ),
+        },
+        extra=vol.PREVENT_EXTRA,
+    ),
+    _check_sensor_schema,
+)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up SAJ sensors."""
+    import pysaj
```

Comment:
Please move 3rd party imports to the top of the module.

Reply:
Done.